### PR TITLE
Debian12 Bookworm for RPi in ci

### PR DESCRIPTION
**Pull request by openoms**
related: https://github.com/raspiblitz/raspiblitz/issues/4215

### CODE REVIEW

**Comment by openoms** [comment]
Can test on RPi4 and RPi5 with the image in the github action artifact 
* need to be logged in to github to see it:
* click the latest successful build in https://github.com/raspiblitz/raspiblitz/actions/workflows/arm64-rpi-lean-image.yml?query=branch%3Arpi-bookworm+is%3Asuccess
* scroll down to download the artifact (a .zip file containing the image)

**Comment by openoms** [comment]
Rebased after the latest merges.

**Comment by openoms** [comment]
Tried the image on the RPi4, there are a few errors (will continue to edit this comment):

During the build (https://github.com/raspiblitz/raspiblitz/actions/runs/6912833281/job/18809024171, [5_Run the build script.txt](https://github.com/raspiblitz/raspiblitz/files/13426617/5_Run.the.build.script.txt)):
* failed to build grpcio
* logrotate: unrecognized service
* rsyslog: unrecognized service

Booting:
* the LCD display comes to a dialog window: `Please enter new username:`
* not connectable via ssh

**Comment by freerko** [comment]
Thanks @openoms for the build! I've got a RP5 and want to test this :)

failed to build grpcio: Is it necessary to install it via PIP, i.e. building it manually, can't we just install the Debian package python3-grpcio and done?

rsyslog and logrotate: Debian 12 switched to systemd-journald for logging by default and thus removed rsyslog and logrotate but sounds like we can just install the packages: https://wiki.debian.org/Rsyslog and https://kifarunix.com/enable-rsyslog-logging-on-debian-12/

**Comment by openoms** [comment]
thanks @freerko I will continue iterate here, please let us know of anything needing to be fixed. There are few issues listed which stood out from the Debian12 release notes here also: https://github.com/raspiblitz/raspiblitz/issues/4215

re grpcio: we should use the pip package, probably it needs an update: 

https://github.com/raspiblitz/raspiblitz/blob/dc2c4c9044a0306a42d7b7ea99c0ad6c00cafa98/build_sdcard.sh#L402

rsyslog: since we alredy have the config for it`sudo apt install rsyslog` should do the job

File: build_sdcard.sh [hunk 1]
```diff
@@ -399,7 +399,7 @@ update-alternatives --install /usr/bin/pip pip /usr/bin/pip3 1
 # 3. Nyx
 # setuptools
 sudo -H python3 -m pip install --upgrade pip
-sudo -H python3 -m pip install grpcio==1.42.0 googleapis-common-protos==1.53.0 toml==0.10.2 j2cli==0.3.10 requests[socks]==2.21.0 protobuf==3.20.1 pathlib2==2.3.7.post1
+sudo -H python3 -m pip install grpcio==1.59.3 googleapis-common-protos==1.61.0 toml==0.10.2 j2cli==0.3.10 requests[socks]==2.31.0 protobuf==4.25.1 pathlib2==2.3.7.post1
```
Comment (openoms) [inline build_sdcard.sh:410]: updating the python deps here due to a build error on Debian12  FYI @rootzoll @fusion44

**Comment by openoms** [comment]
The latest RPi image from the actions [here](https://github.com/raspiblitz/raspiblitz/actions/runs/6948813270?pr=4238) is running recovery on my RPi4 - would worth a try on the RPi5 now @freerko

**Comment by freerko** [comment]
The RPi5 booted from the latest image of the actions, via SSH I get the "Waiting for HDD/SSD" screen. How can I quit the screen back to the CLI? CTRL+C terminates the SSH session.

The 3.5 display hat doesn't work. Tomorrow I'll get a new SSD to test this further.

**Comment by openoms** [comment]
> The RPi5 booted from the latest image of the actions, via SSH I get the "Waiting for HDD/SSD" screen. How can I quit the screen back to the CLI? CTRL+C terminates the SSH session.

CTRL+C should taking you to the prompt. Try to open in annew terminal window or use tmux

> The 3.5 display hat doesn't work

Is there anything on the screen? Is this the same LCD used for the RPi4?

**Comment by openoms** [comment]
Merging these changes for the RC1

**Comment by openoms** [comment]
Will continue to test the images from the dev branch as described here: https://github.com/raspiblitz/raspiblitz/tree/dev/ci#ready-made-images-for-arm64-rpi

**Comment by freerko** [comment]
I've found two old USB HDD drives, one with 1TB and one with 500GB. Both didn't work for a clean new setup: When they are formated as FAT or NTFS the RaspiBlitz recognized them and via SSH I can select a fresh install. RaspiBlitz then tries to format it to ext4 and after about 10s its rebooting:

```
Running Format: (ext4) (sda)                                                                                                                                                                                   
# DATA DRIVE - FORMATTING to EXT4 layout (old)
# Unmounting all partitions of this device
# Cleaning /etc/fstab from d69193c5-903a-4740-8ecf-0ba3023bfd4b
# Make sure system drives are unmounted ..
# Wiping all partitions (sfdisk/wipefs)
# Creating the one big partition

Broadcast message from root@raspberrypi (Wed 2023-11-22 13:37:46 GMT):

The system will reboot now!


Broadcast message from root@raspberrypi (Wed 2023-11-22 13:37:46 GMT):

The system will reboot now!

Connection to 192.168.178.49 closed by remote host.
Connection to 192.168.178.49 closed.
```
Looks like this is already a known issue in #4231. I am using the new official Raspberry Pi power supply with 5A at 5V, so I guess this shouldn't be a power issue. Both disks are pretty old HDDs but they both can be mounted and formated on Linux and Windows laptops without issues, so I assume it is not a hardware issue.
After the reboot I can select FRESHSETUP again which wants to format the drive again, which results in another reboot after about 10s and so on... Looks like my new SSD will arrive tomorrow, then I will test the new setup again.

With the SSD of my old RPi4 RaspiBlitz 1.8.0 the recovery/update was possible. During the recovery bitcoind would stop complaining about a corrupt peers.dat, but after I deleted it "service bitcoind start" worked fine.

The screen is just white. It works fine on the RPi4 RaspiBlitz 1.8.0. Probably this is an issue with Wayland instead of X-Server as the new window server in Debian Bookworm? See https://github.com/goodtft/LCD-show/issues/348. Using raspi-config to change to X-Server (openbox) doesn't change anything for me.

**Comment by neversleep404** [comment]
I currently have a Raspberry Pi 4 running RaspiBlitz 1.10.0. Tomorrow, I will receive a Raspberry Pi 5 and am planning to test the latest build. Is it safe to use the SSD from the Pi 4 for this purpose? I also have a one-week-old backup of the SSD.

**Comment by openoms** [comment]
> I currently have a Raspberry Pi 4 running RaspiBlitz 1.10.0. Tomorrow, I will receive a Raspberry Pi 5 and am planning to test the latest build. Is it safe to use the SSD from the Pi 4 for this purpose? I also have a one-week-old backup of the SSD.

using an old disk should work,but if you are unsure rather wait for others to try first.

Never restore an outdated copy of your channel database as it will lead to force closures of the channels.

**Comment by neversleep404** [comment]
Running raspiblitz-arm64-rpi-image-2023-11-23-9f23886 on:

- Pi 5 8GB
- Raspberry Pi Active Cooler
- LCD - 3.5" RPi Display, GPIO connection, XPT2046 Touch Controller
- 2TB Samsung SSD (from my RaspiBlitz v1.10.0 Pi 4 8GB)

with:

- Core Lightning Node
- CLBOSS

Setup/Migration lasted over an hour. Compiling CL and CLBOSS took a long time. 
The System runs now stable with 92% CPU load and a temperature of 58°C.

It has the same transaction error on the RaspiBlitz homescreen (An error occurred: Unknown error. The response was: 500 Internal Server Error), but that is a different, minor issue with my configuration.

**Pull request merged**